test/integration/targets/iam_user/tasks/main.yml

```
@@ -215,6 +399,29 @@
       that:
         - iam_user_info.iam_users | length == 0
 
+  - name: remove group
+    iam_group:
+      name: '{{ test_group }}'
+      state: absent
+    register: iam_group
+
+  - name: assert that group was removed
+    assert:
+      that:
+        - iam_group.changed
+        - iam_group
+
+  - name: re-remove ansible (idempotency)
+    iam_user:
+      name: AnsibleTestUser
```

Thanks so much for adding these back into a PR.

I don't see the test to ensure that deleting the user with an attached policy works. Also, should this be
```suggestion
      name: '{{ test_user }}'
```
Or was this task meant to be
```
  - name: re-remove group (idempotency)
    iam_group:
      name: '{{ test_group }}'
      state: absent
    register: iam_group
```
The task name could be clarified too.

---

I've added some more tests but I'm uncertain how clear it is by that point (other than because the task name says so) that the user does indeed still have the policy attached - it's not super apparent from the test flow or in any of the return data available at that point that I saw.

---

Thanks, that looks like exactly what I was thinking (the tests were creating "{{ test_user }}" but before the last commit then there weren't iam_user tasks with that var using `state: absent`). If you want to make it more explicit, since the flow isn't the most obvious, you could always get info about the user and assert that there's still an attached policy before using `state: absent`.

---

I did try that but attached policy info is not returned by any of these modules - even when we add/remove policies in the tests we can only assert that the result is changed.  The only place we call `list_attached_user_policies` is when comparing what policies to add or remove in `iam_user.py`, we never actually return the data anywhere.

---

Ooh.... wow, okay. I had no idea.